internal/app/machined/pkg/adapters/perf/cpu.go

```diff
@@ -41,8 +40,22 @@ func (a cpu) Update(stat *procfs.Stat) {
 		}
 	}
 
-	translateListOfCPUStat := func(in []procfs.CPUStat) []perf.CPUStat {
-		return slices.Map(in, translateCPUStat)
+	translateListOfCPUStat := func(in map[int64]procfs.CPUStat) []perf.CPUStat {
```

Comment:
why did we drop `slices.Map`?

Reply:
because update to prometheus/procfs changed the type for CPU stats:

```go
[]procfs.CPUStat
```

became

```go
map[int64]procfs.CPUStat
```

This I guess makes more sense, as some CPUs might be missing. 

But at the same time we can't change our resources/APIs, so we need to convert back to slice, which is tricky in this case.

Reply:
Will probably need a way to show that CPU Core data is missing.

Reply:
we keep it same as way as before - it will be all zeroes (that's what procfs would generate before)